react/features/remote-video-menu/components/native/RemoteVideoMenu.js

```diff
@@ -99,10 +116,12 @@ class RemoteVideoMenu extends Component<Props> {
  *  }}
  */
 function _mapStateToProps(state, ownProps) {
-    const { id } = ownProps.participant;
+    const { participant } = ownProps;
 
     return {
-        _participantDisplayName: getParticipantDisplayName(state, id)
+        _avatarURL: getAvatarURL(participant),
+        _participantDisplayName: getParticipantDisplayName(
```

Comment:
This is a great question. The local reason is that the ```getParticipantDisplayName``` function retrieves the participant from state using the ID, while the ```getAvatarURL``` uses a full participant object, therefore it doesn't need a state reference. I prefer passing full objects instead of just IDs, so I'd prefer the ```getAvatarURL``` way of doing it, but the ```getParticipantDisplayName``` function is used in many places of the application and I'm not sure if the nearby participant object is always a full object, not just a constructed dummy one with an ID, so I'm hesitant to change the signature.

Reply:
Never mind, I decided to change the signature anyhow, it's here in a separate commit.